aspnetcore/client-side/dotnet-interop/index.md

```diff
@@ -0,0 +1,948 @@
+
+* Encapsulate additional logic.
+* Manually map types.
+* Reduce the number of objects or calls crossing the interop boundary.
+* Manually map static calls to instance methods.
+
+## Loading JavaScript declarations
+
+JS declarations which are intended to be imported with `[JSImport]` are typically loaded in the context of the same page or JS host that loaded .NET WebAssembly. This can be accomplished with:
+
+* A `<script>...</script>` block declaring inline JS.
+* A script source (`src`) declaration (`<script src="./some.js"></script>`) that loads an external JS file (`.js`).
+* A JS ES6 module (`<script type='module' src="./moduleName.js"></script>`).
+* A JS ES6 module loaded using <xref:System.Runtime.InteropServices.JavaScript.JSHost.ImportAsync%2A?displayProperty=nameWithType> from .NET WebAssembly.
+
+Examples in this article use <xref:System.Runtime.InteropServices.JavaScript.JSHost.ImportAsync%2A?displayProperty=nameWithType>. When calling <xref:System.Runtime.InteropServices.JavaScript.JSHost.ImportAsync%2A>, client-side .NET WebAssembly requests the file using the `moduleUrl` parameter, and thus it expects the file to be accessible as a static web asset, much the same way as a `<script>` tag retrieves a file with a `src` URL. For example, the following C# code within a WebAssembly Browser App project maintains the JS file (`.js`) at the path `/wwwroot/scripts/ExampleShim.js`:
+
+```csharp
+await JSHost.ImportAsync("ExampleShim", "/scripts/ExampleShim.js");
+```
+
+Depending on the platform that's loading WebAssembly, a dot-prefixed URL, such as `./scripts/`, might refer to an incorrect subdirectory, such as `/_framework/scripts/`, because the WebAssembly package is initialized by framework scripts under `/_framework/`. In that case, prefixing the URL with `../scripts/` refers to the correct path. Prefixing with `/scripts/` works if the site is hosted at the root of the domain. A typical approach involves configuring the correct base path for the given environment with an HTML `<base>` tag and using the `/scripts/` prefix to refer to the path relative to the base path. Tilde notation `~/` prefixes aren't supported by <xref:System.Runtime.InteropServices.JavaScript.JSHost.ImportAsync%2A?displayProperty=nameWithType>.
+
+> [!IMPORTANT] 
+> If JS is loaded from an ES6 module, then `[JSImport]` attributes must include the module name as the second parameter. For example, `[JSImport("globalThis.callAlert", "ExampleShim")]` indicates the imported method was declared in an ES6 module named "`ExampleShim`."
```

Comment:
What's the exact change request?

Reply:
**Is there better terminology that would still distinguish the _specific_ module pattern needed for ImportAsync to work successfully?**  As far as I could tell ImportAsync will not work unless I declared the *.js as what I've always seen distinguished from other module patterns such as AMD as an "ES6 module", but I don't doubt it might be the wrong terminology.  Point being, ImportAsync won't happily accept anything, and it would be nice to convey in some way which **specific** module pattern it requires.  I find it a little frustrating when documentation uses the term "module" without specificity, only to find it is more specific than implied.  It could be argued that there's really only one true "module" syntax in official JS, but it kind of disregards the reality of the ecosystem of module patterns.  You will find other documentation that uses the term "module", but is implicitly referring to another module pattern that was the more pervasive pattern at the time, and for the same reason it can be a hurdle/time sink without knowing the specificity.

Reply:
my point was that it only requires a second parameter, when you loaded it via `JSHost.ImportAsync`
The fact that it needs to be ES6 in order to be loaded via `JSHost.ImportAsync` is a separate (useful) information.